### PR TITLE
Remove dead code from Microsoft.CSharp

### PR DESCRIPTION
**Pull request by JonHanna**
The only paths reaching `ExprArrayLength` and `ExprQuestionMark` were removed in #17948 so they can also be deleted, along with code that this in turn makes dead.

Since `ExprArrayLength` obviously related to array lengths, add tests double-checking that still works (it does, because it isn't special-cased like it is with static code and is parsed like any other property).

Also delete all the code identified as dead in #17905 that is really dead (not some constants), not in shared files, and not going to be deleted by #18004.

### CODE REVIEW

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Errors/ErrorFmt.cs [hunk 1]
```diff
@@ -155,10 +140,7 @@ public ErrArg(MethPropWithInst mpwi)
         {
             return new ErrArg(n);
         }
```
Comment (OmarTawfik) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Errors/ErrorFmt.cs]: nit: spacing in the file should be consistent.

**Comment by danmosemsft** [comment]
@VSadov @OmarTawfik PTAL

**Comment by danmosemsft** [comment]
@VSadov @OmarTawfik

**Comment by OmarTawfik** [comment]
@JonHanna thanks for the PR!
@danmosemsft will review it today.

**Comment by OmarTawfik** [comment]
>                     Debug.Assert(false, "Bad symbol kind");

nit: might be useful to include the SIMKIND in the assert message.

---
Refers to: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Errors/UserStringBuilder.cs:521 in 36743c2. [](commit_id = 36743c2f91be720301534b72cc607723c5928fa4, deletion_comment = True)

**Comment by karelz** [comment]
Given that this is larger change, it would be perfectly fine to wait with merging it after 5/10 - see https://github.com/dotnet/corefx/issues/17619#issuecomment-296872132

**Comment by danmosemsft** [comment]
Master is open again. Do you want to rebase and continue?

**Comment by JonHanna** [comment]
I shall. I don't have a lot of time right now, but I should at least manage a rebase some time today.

**Comment by danmosemsft** [comment]
@VSadov can this be merged?

**Pull request merged**